units/dim.cpp

```diff
@@ -16,6 +16,7 @@ std::unordered_map<std::string, DimId> Dim::builtin_ids{
     {"group", Dim::Group},
     {"<invalid>", Dim::Invalid},
     {"position", Dim::Position},
+    {"pulse-time", Dim::PulseTime},
```

Comment:
I forget, can you remind me how do we take care of the case where someone calls it `"pulse_time"` instead? Will it then not be in the known units? Is it then just a matter of having to set the unit by hand?

Reply:
For unknown dimension labels (I assume that is what you mean, units are something else) it is added to a map. It works just as well, but there is a lock in the string to/from dim conversion.